### PR TITLE
Add a timed loader class

### LINKED ISSUE #329: Finalize and replace app check loading logic in -[GIDSignIn authorizationRequestWithOptions:completion:]`
The current implementation dispatches async after waiting 1/2 second to ensure that an activity spinner is displayed to avoid flashing. We should update the logic to instead do the following:

1. If the call to app check completes before some short threshold is met, just carry on without any loading screen.
2. If that threshold is exceeded, then display a loading screen for at least 1/2 second.
3. After displaying the loading screen for 1/2, complete when callback is received from app check.

### PR DESCRIPTION
**Pull request by mdmathias**
GIDTimedLoader encapsulates the logic to handle presenting an activity indicator when fetching the app check token.

Fixes #329

### CODE REVIEW

File: GoogleSignIn/Sources/GIDSignIn.m [hunk 1]
```diff
@@ -634,35 +639,29 @@ - (void)authorizationRequestWithOptions:(GIDSignInInternalOptions *)options comp
...
-            }
-            [activityVC.activityIndicator stopAnimating];
-            [activityVC dismissViewControllerAnimated:YES completion:nil];
-            completion(nil, error);
-            return;
+      UIViewController *presentingVC = options.presentingViewController;
+      if (!_timedLoader) {
+        _timedLoader = [[GIDTimedLoader alloc] initWithPresentingViewController:presentingVC];
+      }
+      if (_timedLoader.animationStatus != GIDTimedLoaderAnimationStatusAnimating) {
```
Comment (yalkhder) [inline GoogleSignIn/Sources/GIDSignIn.m:646]: Do we need this check if GIDTimedLoader does that check anyway in the implementation of startTiming?

File: GoogleSignIn/Sources/GIDTimedLoader/GIDTimedLoader.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <TargetConditionals.h>
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+typedef NS_ENUM(NSUInteger, GIDTimedLoaderAnimationStatus) {
```
Comment (yalkhder) [inline GoogleSignIn/Sources/GIDTimedLoader/GIDTimedLoader.h:22]: Add declarative comments for enum.

File: GoogleSignIn/Sources/GIDTimedLoader/GIDTimedLoader.m [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                                              completion:nil];
+  });
+}
+
+- (void)stopTimingWithCompletion:(void (^)(void))completion {
+  if (self.animationStatus != GIDTimedLoaderAnimationStatusAnimating) {
+    return;
+  }
+
+  if (self.loadingTimer) {
```
Comment (yalkhder) [inline GoogleSignIn/Sources/GIDTimedLoader/GIDTimedLoader.m:83]: The nil check is not needed here is it?

File: GoogleSignIn/Sources/GIDSignIn.m (hunk 1 above)
Comment (mdmathias) [reply GoogleSignIn/Sources/GIDSignIn.m:646]: Nope!

File: GoogleSignIn/Sources/GIDTimedLoader/GIDTimedLoader.m (hunk 3 above)
Comment (mdmathias) [reply GoogleSignIn/Sources/GIDTimedLoader/GIDTimedLoader.m:83]: Nope!

**Review by yalkhder — APPROVED** [review]

**Review by yalkhder — APPROVED** [review]

**Pull request merged**